builtin/providers/aws/resource_aws_instance.go

```diff
@@ -129,9 +129,11 @@ func resourceAwsInstance() *schema.Resource {
 			},
 
 			"network_interface_id": &schema.Schema{
-				Type:     schema.TypeString,
-				Computed: true,
-				Optional: true,
+
+				Type:          schema.TypeString,
```

Comment:
OK, LGTM. I guess there are not many OS that object to hotplugging NICs after boot